src/augmented_dickey_fuller.jl

```diff
@@ -0,0 +1,219 @@
+# augmented_dickey_fuller.jl
+# (Augmented) Dickey-Fuller unit root test
+#
+# Copyright (C) 2017   Benjamin Born
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export ADFTest
+
+immutable ADFTest <: HypothesisTest
+    n               ::Int               # number of observations
+    deterministic   ::AbstractString    # deterministic terms included in regression
```

Comment:
It would be great if you could use symbols, I think that's the better choice here. Stefan's explanation is good but deeper than one really needs to know to use them for everyday tasks. `Symbol` is an interesting type in that it plays a central role in Julia's AST, which is made up of `Expr`s and `Symbol`s, but `Symbol`s can be used more generally; for example, we use them in this package to specify the types for confidence intervals and in DataFrames we use them for column names. You can think of them sort of like strings that cannot be decomposed further. (Strings can be decomposed into `Char`s.)